### PR TITLE
Style guide

### PR DESCRIPTION
**Pull request by ifrit98**
* Port Bittensor style guide and development workflow over for all subnets.

### CODE REVIEW

File: contrib/DEVELOPMENT_WORKFLOW.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      - [Finishing a Release Branch](#finishing-a-release-branch)
+      - [Creating a Hotfix Branch](#creating-a-hotfix-branch)
+      - [Finishing a Hotfix Branch](#finishing-a-hotfix-branch)
+  - [Continuous Integration (CI) and Continuous Deployment (CD)](#continuous-integration-ci-and-continuous-deployment-cd)
+  - [Versioning and Release Notes](#versioning-and-release-notes)
+  - [Pending Tasks](#pending-tasks)
+
+## Main Branches
+
+Bittensor's codebase consists of two main branches: **master** and **staging**.
```
Comment (steffencruz) [inline contrib/DEVELOPMENT_WORKFLOW.md:26]: Bittensor's codebase consists of two main branches: **main** and **staging**.

File: contrib/DEVELOPMENT_WORKFLOW.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+**master**
```
Comment (steffencruz) [inline contrib/DEVELOPMENT_WORKFLOW.md:28]: main

File: contrib/DEVELOPMENT_WORKFLOW.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+- Branch off from: `staging`
+- Merge back into: `staging`
+- Naming convention: `feature/<ticket>/<descriptive-sentence>`
+
+Feature branches are used to develop new features for upcoming or future releases. They exist as long as the feature is in development, but will eventually be merged into `staging` or discarded. Always delete your feature branch after merging to avoid unnecessary clutter.
+
+### Release Branches
+
+- Branch off from: `staging`
+- Merge back into: `staging` and then `master`
```
Comment (steffencruz) [inline contrib/DEVELOPMENT_WORKFLOW.md:47]: main

File: contrib/DEVELOPMENT_WORKFLOW.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+- Naming convention: `release/<version>/<descriptive-message>/<creator's-name>`
+
+Release branches support the preparation of a new production release, allowing for minor bug fixes and preparation of metadata (version number, configuration, etc). All new features should be merged into `staging` and wait for the next big release.
+
+### Hotfix Branches
+
+General workflow:
+
+- Branch off from: `master` or `staging`
```
Comment (steffencruz) [inline contrib/DEVELOPMENT_WORKFLOW.md:56]: main

File: contrib/DEVELOPMENT_WORKFLOW.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+- Merge back into: `staging` then `master`
```
Comment (steffencruz) [inline contrib/DEVELOPMENT_WORKFLOW.md:57]: main

Comment (steffencruz) [reply contrib/DEVELOPMENT_WORKFLOW.md:57]: I'm going to stop saying main now.

**Review by steffencruz — APPROVED** [review]
LGTM

File: contrib/DEVELOPMENT_WORKFLOW.md (hunk 1 above)
Comment (ifrit98) [reply contrib/DEVELOPMENT_WORKFLOW.md:26]: Ahh, thanks!

**Pull request merged**